enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,545 @@
+
+### Implementation Details
+
+> Note: The following sub-headers may frequently refer to OLM 1.x terminology. It's recommended to read through the ["Common Terminology"](#Common-Terminology) header for those less familiar with recent OLM developments.
+
+#### FeatureGate
+
+The components outlined in the [Proposal](#Proposal) section will be delivered as a tech preview feature for openshift clusters, and therefore this mechanism won't be installed by default.
+
+An admin can enable the "TechPreviewNoUpgrade" feature set before cluster creation as outlined in the [Workflow Description](#Workflow-Description) section:
+
+> Note: admins can only enable POs through the TechPreviewNoUpgrade feature set, and therefore enabling the CustomNoUpgrade feature set requirement performing some manual steps.
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: FeatureGate
+metadata:
+  name: cluster
+spec:
+  featureSet: TechPreviewNoUpgrade
+```
+
+Enabling the tech preview feature set will result in the installation of the PO components in the cluster. This requires adding the "release.openshift.io/feature-gate" annotation key to the PO's downstrema CVO manifests, and setting that key's value to "TechPreviewNoUpgrade".
+
+#### The PO Manager (POM)
+
+The PO manager would abstract the process of sourcing registry+v1 bundles from catalog(s), choosing the "best bundle" to install for each desired package, and ensuring those packages get installed on the cluster.
+
+In order to determine the set of bundles that need to be installed, the PO manager can interact with the default, `redhat-operators` CatalogSource, and use OLM's existing registry gRPC APIs to list the available bundles from those catalog contents.
+
+> The OLM marketplace component is now an optional CVO component. Admins can disable this component using the component capabilities functionality introduced in previous 4.x releases. Available workarounds include admins creating their own CatalogSource resources. This behavior may require explicit documentation, or expanding the phase 0 design scope to be able to accommodate all the catalog sources in the cluster.
+
+After determining the set of bundles that need to exist in order to satisfy admin package intent, the PO manager's application layer will need to attempt to install those bundles. The management of those bundle contents will be delegated to OLM's rukpak layer, and the POM component would be responsible for generating an underlying BundleDeployment resource for the bundle that was selected. This underlying BundleDeployment resource will contain owner references to the parent PlatformOperator custom resource the manager is reconciling. The failed deployment of a bundle should be reflected in the aggregate ClusterOperator resource.
```

Comment:
> This underlying BundleDeployment resource will contain owner references to the parent PlatformOperator

What is the scope of a BundleDeployment? Are they namespaced or cluster scoped?

Reply:
Updated some of the language here and made it more explicit this was a cluster-scoped API.